src/Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions/IConnectionInformation.cs

```diff
@@ -12,9 +12,10 @@ public interface IConnectionInformation
         IPEndPoint LocalEndPoint { get; }
 
         PipeFactory PipeFactory { get; }
+        IScheduler InputReaderScheduler { get; }
```

Comment:
An alternative would be to give the Transport control over the scheduler (here I go again). If meaningful for the Transport, it can expose dispatch control via a Transport specific option.